tests/poller.cpp

```diff
@@ -20,9 +19,7 @@ TEST(poller, move_construct_empty)
     std::unique_ptr<zmq::poller_t> a{new zmq::poller_t};
     zmq::poller_t b = std::move(*a);
 
-    ASSERT_EQ(0u, a->size ());
     a.reset ();
-    ASSERT_EQ(0u, b.size ());
```

Comment:
There must be some assertion here. Or maybe the move operations of poller_t should be deleted entirely.

Reply:
It is perfectly valid that unit test does not assert and runs without a crash/exception but if maintainers decide otherwise I am fine with that.

Reply:
I completely agree in principle, but I think in this case there should be some assertion.

This test would accept a move constructor that is a no-op...

But maybe it is fine if proper assertion are added for the non-empty case.

But still, I am not sure if it makes sense to provide move operations at all. If one wants to move a poller, they could use a unique_ptr to a poller_t instead.

Reply:
Everything inside the poller is movable already so it make it easy to keep poller movable too.